packages/open-next/src/adapters/util.ts

```diff
@@ -1,9 +1,60 @@
 import fs from "node:fs";
 import path from "node:path";
+import { request } from "node:https";
 
 export function loadConfig(nextDir: string) {
   const filePath = path.join(nextDir, "required-server-files.json");
   const json = fs.readFileSync(filePath, "utf-8");
   const { config } = JSON.parse(json);
   return config;
 }
+export function loadBuildId(nextDir: string) {
+  const filePath = path.join(nextDir, "BUILD_ID");
+  return fs.readFileSync(filePath, "utf-8").trim();
+}
+
+export interface PrerenderManifest {
+  version: number;
+  routes: {
+    dataRoute: string;
+    srcRoute: string | null;
+    initialRevalidateSeconds: number | boolean;
+  }[];
+  dynamicRoutes: {
+    routeRegex: string;
+    dataRoute: string;
+    fallback: string | null;
+    dataRouteRegex: string;
+  }[];
+  preview: {
+    previewModeId: string;
+    previewModeSigningKey: string;
+    previewModeEncryptionKey: string;
+  };
+}
+
+export function loadPrerenderManifest(nextDir: string) {
+  const filePath = path.join(nextDir, "prerender-manifest.json");
+  const json = fs.readFileSync(filePath, "utf-8");
+  return JSON.parse(json) as PrerenderManifest;
+}
+
+export async function revalidateInBackground(
+  host: string,
+  path: string,
+  preview: string
+) {
+  try {
+    await new Promise<void>((resolve, reject) => {
+      request(`https://${host}${path}`, {
+        method: "HEAD",
+        headers: { "x-prerender-revalidate": preview },
+      })
+        .on("error", (err) => reject(err))
+        .end(() => resolve());
+    });
+  } catch (e) {
+    console.error("Failed to revalidate stale page.", path);
+    console.error(e);
+  }
+}
```

Comment:
Actually no, we don't wait for data here, but just for the request to be sent

Reply:
As you mentionned above, i think we could probably go for a SQS queue here instead of just a head request. This could allow us to have deduplication and error handling